R/kartverket.R

```diff
@@ -0,0 +1,211 @@
+#' Get address info
+#'
+#' Search for Norwegian addresses in
+#' \emph{The Norwegian Mapping Authority's Cadastre and Land Registry}.
+#'
+#' @details
+#' Many users will get by with using the regular `search` parameter, but more
+#' advanced users might want to look into the wide range of possible parameters.
+#' For a list of all accepted query parameters see [create_params_kv_sok()]
+#' or refer to the
+#' [API documentation](https://ws.geonorge.no/adresser/v1/#/default/get_sok)
+#' for further details (in Norwegian only).
+
+#' The function returns a data.frame or list by default. If you set `raw_response`
+#' to `TRUE`, the raw response from the API will be returned instead. Note that the
+#' response will then be returned silently.
+#'
+#' @param search A Norwegian address. For example: 'munkegata 1 trondheim'.
+#' @param crs The desired coordinate reference system for the output. Default
+#'   value is ETRS 89 for latitude and longitude data (EPSG:4258).
+#' @param ... Additional arguments passed to the API call. See details.
+#' @inheritParams get_entity
+#'
+#' @return data.frame or list
+#' @export
+#' @examples
+#'
+#' # A simple query
+#' df <- get_address_info('munkegata 1 trondheim')
+#'
+#' # Use the advanced parameters to be more specific
+#' df <- get_address_info(address_name = 'Munkegata', address_number = 1,
+#'          mun_name = 'Trondheim')
+#'
+#' \dontrun{
+#' # Multiple adresses
+#' res <- get_address_info(c('munkegata 1 trondheim', 'tromsø gate'))
+#'
+#' # A more advanced query
+#' res <- get_address_info(c('munkegata 1 trondheim', 'munkegata 1'),
+#'    mun_code = c(NULL, '5001'))
+#'
+#' }
+get_address_info <- function(search = NULL, crs = 4258, ..., raw_response = FALSE) {
+  if (!is.null(search)) {
+    dl <- mapply(get_address_info_safe, search = search,
+                 crs = crs, ..., raw_response = raw_response,
+                 SIMPLIFY = FALSE, USE.NAMES = FALSE)
+  } else {
+    dl <- mapply(get_address_info_safe, crs = crs,
+                 ..., raw_response = raw_response,
+                 SIMPLIFY = FALSE, USE.NAMES = FALSE)
+  }
+  if (raw_response) return(invisible(dl))
+  if (length(dl) == 1) dl <- dl[[1]]
+  dl <- dl[lengths(dl) != 0]
+  dl
+}
+
+#' Get address info (single)
+#' @inheritParams get_address_info
+#' @noRd
+get_address_info_single <- function(search = NULL, crs = 4258,
+                                    ..., raw_response = FALSE) {
+  resp <- request_kartverket('sok', search, crs = crs, ...)
+  parsed <- parse_response(resp, simplifyVector = TRUE)
```

Comment:
Good question. We could certainly do that. I left it as is since it sort of makes sense that the geometry columns are bundled together and base R now has good support for data.frames within data.frames.  The difference between selecting e.g `df$representasjonspunkt$lon` and `df$representasjonspunkt_lon` is also minimal. But certainly open to the idea of returning a completely flat table.